src/ucp/api/ucp_def.h

```diff
@@ -612,11 +612,13 @@ typedef ucs_status_t (*ucp_am_callback_t)(void *arg, void *data, size_t length,
  * The callback is always called from the progress context, therefore calling
  * communication progress routines from this callback is not allowed. It is
  * recommended to define callbacks with relatively short execution time to
- * avoid blocking of communication progress.
+ * avoid blocking of communication progress. The callback is thread safe with
+ * respect to the worker it is registered on.
  *
  * @param [in]  arg           User-defined argument.
  * @param [in]  header        User defined active message header. Can be NULL.
- * @param [in]  header_length Active message header length in bytes.
+ * @param [in]  header_length Active message header length in bytes. Can be 0
```

Comment:
I think what you are trying to say: " if @a header is NULL, this value is ignored" ?

Reply:
I have to pass some value in any case, maybe it is better to define 0 explicitly. Otherwise some user may check header_length for 0 and try to access header if it is not NULL

Reply:
It should be phrased as `If this value is 0, the @a header pointer is undefined and should not be accessed`
(a->b is not same as b->a)